compiler/tests-common/tests/org/jetbrains/kotlin/jvm/compiler/ir/AbstractIrLoadJavaTest.kt

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.jvm.compiler.ir
+
+import org.jetbrains.kotlin.config.CompilerConfiguration
+import org.jetbrains.kotlin.config.JVMConfigurationKeys
+import org.jetbrains.kotlin.jvm.compiler.AbstractLoadJavaTest
+
+abstract class AbstractIrLoadJavaTest : AbstractLoadJavaTest() {
```

Comment:
Let's inherit from `AbstractLoadJavaWithFastClassReadingTest` instead. This might be a bit confusing, but "fast class reading" (loading classes directly from class files by ASM) is used in the _compiler_. The other class reading mode (loading classes by querying `PsiClass` constructed by IntelliJ) is used in the _IDE_ and is tested in `AbstractLoadJavaTest`.

For the IR backend scenario, I think it makes more sense to test the compiler behavior.

Reply:
Done. Thanks again for the context, that sounds right to me as well.